src/blockchainApiConnection/BlockchainApiConnection.ts

```diff
@@ -38,9 +38,7 @@ export async function buildConnection(
     provider,
     types: CUSTOM_TYPES,
   })
-  const bc = new Blockchain(api)
-  await bc.ready
-  return bc
+  return new Blockchain(api)
```

Comment:
Shouldn't this be new `Blockchain(api).ready`?

Or do we not need the `ready` now since you removed it from the blockchain constructor? 

`this.ready = this.errorHandler.ready`

Reply:
yeah, no more .ready needed, this was just to initialise the error handler

Reply:
Ok, good to know.